zokrates_analysis/src/constant_argument_checker.rs

```diff
@@ -72,6 +72,28 @@ impl<'ast, T: Field> ResultFolder<'ast, T> for ConstantArgumentChecker {
         }
     }
 
+    fn fold_field_expression(
```

Comment:
I had to move this check here from propagation, this was an unrelated issue to this PR but one of my assembly tests was failing because of it. We had this check in propagation which would fail in this case:

```
def test<N>() -> field {
    field mut a = 1;
    for u32 i in 0..N {
        a = a * 2;
    }
    field b = 2**N; //  Non-constant exponent `N_1` detected during static analysis
    return b;
}
```

The code above will error out on develop. It seems we call propagation a few times but in one of the calls `N_1` is not available as a constant. Also, it seems like this is the correct place to put this check as we do check shifts in this module.